spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -104,13 +107,20 @@ public class RewriteManifestsSparkAction
             TableProperties.MANIFEST_TARGET_SIZE_BYTES,
             TableProperties.MANIFEST_TARGET_SIZE_BYTES_DEFAULT);
 
-    // default the staging location to the metadata location
+    // default the output location to the metadata location
     TableOperations ops = ((HasTableOperations) table).operations();
     Path metadataFilePath = new Path(ops.metadataFileLocation("file"));
-    this.stagingLocation = metadataFilePath.getParent().toString();
+    this.outputLocation = metadataFilePath.getParent().toString();
 
     // use the current table format version for new manifests
     this.formatVersion = ops.current().formatVersion();
+
+    boolean snapshotIdInheritanceEnabled =
+        PropertyUtil.propertyAsBoolean(
+            table.properties(),
+            TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED,
+            TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED_DEFAULT);
+    this.shouldStageManifests = formatVersion == 1 && !snapshotIdInheritanceEnabled;
```

Comment:
What if we made change here, https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadata.java#L482-L482, like this, so that we don't have to keep check the format version? 
```
  public boolean propertyAsBoolean(String property, boolean defaultValue) {
    boolean value = PropertyUtil.propertyAsBoolean(properties, property, defaultValue);
    if (property == TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED) {
       if (formatVersion == 1) {
           return value;
        } else {
           return true;
        }
     }
     return value;
  }
```

Reply:
The problem that we are not using `TableMetadata` in this case. We are using `PropertyUtil` on public table properties instead. There, we don't have the format version, unless we pass it.

I think the proper way to fix this would be to offer something like `TableConf`, that would have dedicated method for properties.